packages/ui-kit/src/components/rss-feeds.js

```diff
@@ -1,121 +1,78 @@
-    if (process.env.NODE_ENV !== 'production') {
-      return <ContentNotifications.Error>{message}</ContentNotifications.Error>;
-    }
-
+  if (!props.dataSources) {
+    const message = (
+      <ContentNotifications.Error>
+        Must pass a source to RssFeeds component
+      </ContentNotifications.Error>
+    );
     throw new Error(message);
   }
 
-  async function fetcher(...args) {
-    const rssParser = new Parser();
-    const feed = await rssParser.parseURL(args[0]);
-    return feed;
-  }
-
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  const { data, error } = useSWR(props.url, fetcher);
+  const { data, error } = useSWR(props.dataSources, fetcher);
 
   if (error) {
     console.log(error);
-    return (
+    const message = (
       <ContentNotifications.Error>
-        Error Loading RSS feed from {props.url}
+        Error Loading Data from {props.dataSources}
       </ContentNotifications.Error>
     );
+    throw new Error(message);
   }
 
-  return (
-    <div>
-      {!data ? (
-        <LoadingSpinner size="s">{'Loading feeds'}</LoadingSpinner>
-      ) : (
-        <Table>
-          <thead>
-            <tr>
-              <th colSpan="2">{props.title ? props.title : data.title}</th>
-            </tr>
-          </thead>
-          <tbody>
-            {data.items.map((item, index) => (
-              <tr key={`${item.title}${index}`}>
-                <td
-                  css={css`
-                    width: ${dimensions.widths.rssFeedDateWidth};
-                  `}
-                >
-                  <DateWrapper>
-                    <Link
-                      href={item.link}
-                      css={css`
-                        text-decoration: none;
-                      `}
-                    >
-                      {moment(item.pubDate).format('D MMMM YYYY')}
-                    </Link>
-                  </DateWrapper>
-                </td>
-                <td>{item.title}</td>
-              </tr>
-            ))}
-          </tbody>
-        </Table>
-      )}
-    </div>
-  );
+  if (data) {
+    const oldestEntriesOfFeeds = data.reduce((list, feed) => {
+      return [...list, feed[feed.length - 1]];
+    }, []);
+
+    // we use the newest of the oldest entry of each feed as the last entry in the release note list
+    const lastEntryOfList = oldestEntriesOfFeeds.reduce(
+      (currentOldestEntry, entry) => {
+        return entry.pubDate > currentOldestEntry.pubDate
+          ? entry
+          : currentOldestEntry;
+      }
+    );
+
+    const limitedEntries = data.flat().reduce((list, entry) => {
+      return entry.pubDate >= lastEntryOfList.pubDate
+        ? [...list, entry]
+        : [...list];
+    }, []);
+
+    const tableData = limitedEntries.sort((dateOne, dateTwo) => {
+      return new Date(dateTwo.pubDate) - new Date(dateOne.pubDate);
+    });
+
+    return <RssFeedTable data={tableData} />;
```

Comment:
I would prefer to do the "transformation/aggregation" of the data into a separate function, to not bloat too much the component.

Reply:
I moved it now into a new function outside of the component 👍